### PR TITLE
Add fp32 test to nightly run

### PR DESCRIPTION
**Pull request by natke**

### CODE REVIEW

**Review by RyanUnderhill — APPROVED** [review]

File: test/python/test_onnxruntime_genai_e2e.py [hunk 1]
```diff
@@ -53,5 +53,7 @@ def run_model(model_path: str | bytes | os.PathLike):
     for model_name in ["microsoft/phi-2"]:
         with tempfile.TemporaryDirectory() as temp_dir:
             device = "cuda" if og.is_cuda_available() else "cpu"
-            download_model(temp_dir, device, model_name)
+            download_model(temp_dir, device, model_name, int4)
```
Comment (baijumeswani) [inline test/python/test_onnxruntime_genai_e2e.py:56]: I am a little confused. How does python know that `int4` is a `str`? Should it not be `"int4"`?

File: test/python/test_onnxruntime_genai_e2e.py [hunk 2]
```diff
@@ -53,5 +53,7 @@ def run_model(model_path: str | bytes | os.PathLike):
...
             run_model(temp_dir)
+            download_model(temp_dir, "cpu", model_name, fp32)
```
Comment (baijumeswani) [inline test/python/test_onnxruntime_genai_e2e.py:58]: Do you not want to execute the model in this test?

File: test/python/test_onnxruntime_genai_e2e.py (hunk 1 above)
Comment (natke) [reply test/python/test_onnxruntime_genai_e2e.py]: Fixed

File: test/python/test_onnxruntime_genai_e2e.py (hunk 2 above)
Comment (natke) [reply test/python/test_onnxruntime_genai_e2e.py]: Fixed

Comment (baijumeswani) [reply test/python/test_onnxruntime_genai_e2e.py]: I think you can do this:

```python
    for precision in ["int4", "fp32"]:
        with tempfile.TemporaryDirectory() as temp_dir:
            device = "cuda" if og.is_cuda_available() else "cpu"
            download_model(temp_dir, device, model_name, precision)
            run_model(temp_dir)
```

Comment (natke) [reply test/python/test_onnxruntime_genai_e2e.py]: Updated

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**